### PR TITLE
EOS-21241 : Main-SMS2-Build 631-Support bundle generation failed for sspl

### PR DESCRIPTION
**Pull request by satish-seagate**
Signed-off-by: Satish Darade <satish.darade@seagate.com>

## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [x] Unittest updated and all unittests are passing?

* [x] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [x] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

**Review by thavanathan — APPROVED** [review]

**Review by sandeepsgt — APPROVED** [review]

**Review by jjxsg — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate [hunk 1]
```diff
@@ -64,13 +65,16 @@ class SSPLBundleGenerate(object):
         self.enclosure_wwn = "NA"
 
     def generate_sspl_support_bundle(self, parser):
-        if os.path.isfile(SSPL_SB_REQUEST):
-            msg = "One SSPL Support Bundle request is ongoing already, skipping current request..!"
+        allow_new_request = validate_sb_request()
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate:68]: Need to update class name to SupportBundle and Generate() can be one of the interface here, renaming generate_sspl_support_bundle() to generate()

File: low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate [hunk 2]
```diff
@@ -278,6 +282,19 @@ def sspl_bundle_cleanup():
     if os.path.exists(SSPL_SB_TMP):
         shutil.rmtree(SSPL_SB_TMP)
 
+def validate_sb_request():
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate:285]: rename to just validate()

**Review by cdeshmukh — COMMENTED** [review]
minor comments, rest looks fine

File: low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate (hunk 1 above)
Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate]: Done

File: low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate (hunk 2 above)
Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate]: Done

**Review by cdeshmukh — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate [hunk 3]
```diff
@@ -63,14 +64,17 @@ class SSPLBundleGenerate(object):
         self.enclosure_log = f"{SSPL_SB_TMP}/enclosure_logs.zip"
         self.enclosure_wwn = "NA"
 
-    def generate_sspl_support_bundle(self, parser):
-        if os.path.isfile(SSPL_SB_REQUEST):
-            msg = "One SSPL Support Bundle request is ongoing already, skipping current request..!"
+    def generate(self, parser):
+        allow_new_request = validate()
+        if not allow_new_request:
+            msg = "SSPL Support Bundle creation in progress. Existing..."
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate:70]: seems some typo here @satish-seagate 
needs 'Exiting' or better 'aborting'

take care in any of your future PR

Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/generate_sspl_bundle/sspl_bundle_generate:70]: Sure Chetan,
Actually, as per ujjwal comments I have made this change. But yes I will take a look into the future patches.
https://github.com/Seagate/cortx-monitor/pull/424#discussion_r653718514

**Pull request merged**